splice_machine/src/main/java/com/splicemachine/derby/vti/ORCRecordIterator.java

```diff
@@ -0,0 +1,246 @@
+package com.splicemachine.derby.vti;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.hive.ql.io.orc.OrcFile;
+import org.apache.hadoop.hive.ql.io.orc.Reader;
+import org.apache.hadoop.hive.ql.io.orc.RecordReader;
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.log4j.Logger;
+
+import com.splicemachine.db.iapi.sql.execute.ExecRow;
+import com.splicemachine.derby.impl.sql.execute.operations.LocatedRow;
+
+/**
+ * This is iterator for the records in the ORC files. It supports either a
+ * single file or Files in a folder. In case of file the next() will return the
+ * rows from one file, and then move to the next file
+ * 
+ * @author jramineni
+ * 
+ */
+
+public class ORCRecordIterator implements Iterable<LocatedRow>,
+Iterator<LocatedRow>, Closeable {
+
+	private static final Logger LOG = Logger.getLogger(ORCRecordIterator.class);
+
+	// The format of the Record to be returned
+	private ExecRow execRow;
+
+	// This is contains the details of the ORC File Record format
+	private StructObjectInspector inspector;
+
+	// Records in Orc File to be processed
+	RecordReader records;
```

Comment:
Is there a reason this is package-scoped? We are letting a lot of state leak here.